FakebookNotifications/FakebookNotifications.Testing/MongoDbTests.cs

```diff
@@ -0,0 +1,255 @@
+            _mockSettings = new Mock<IOptions<NotificationsDatabaseSettings>>();
+            _mockDB = new Mock<IMongoDatabase>();
+            _mockClient = new Mock<IMongoClient>();
+
+            settings = new NotificationsDatabaseSettings()
+            {
+                ConnectionString = "mongodb://localhost:27017",
+                DatabaseName = "NotificationsDb",
+                UserCollection = "User",
+                NotificationsCollection = "Notifications"
+            };
+        }
+
+        //Test for context constructor to see if database is connected to successfully
+        [Fact]
+        public void MongoContextCreationTest()
+        {
+            //Arrange
+            //Setup Context Settings
+            _mockSettings.Setup(s => s.Value).Returns(settings);
+
+            //Setup Mongo Client
+            _mockClient.Setup(c => c.GetDatabase(_mockSettings.Object.Value.DatabaseName, null))
+                .Returns(_mockDB.Object);
+
+            //Act
+            var context = new NotificationsContext(_mockSettings.Object); //create context and test constructor
+
+            //Assert
+            Assert.NotNull(context);
+        }
+
+        //Test to see if the notification collection is successfully read from db
+        [Fact]
+        public void MongoNotificationCollectionTest()
+        {
+            //Arrange
+            //Setup Context Settings
+            _mockSettings.Setup(s => s.Value).Returns(settings);
+
+            //Setup Mongo Client
+            _mockClient.Setup(c => c.GetDatabase(_mockSettings.Object.Value.DatabaseName, null))
+                .Returns(_mockDB.Object);
+
+            //Act
+            var context = new NotificationsContext(_mockSettings.Object); //Create context
+            var notificationCollection = context.Notifications; //Get collection from context
+
+            //Assert 
+            Assert.NotNull(notificationCollection);
+        }
+
+        //Test to see if the user collection is successfully read from db
+        [Fact]
+        public void MongoUserCollectionTest()
+        {
+            //Arrange
+            //Setup Context Settings
+            _mockSettings.Setup(s => s.Value).Returns(settings);
+
+            //Setup Mongo Client
+            _mockClient.Setup(c => c.GetDatabase(_mockSettings.Object.Value.DatabaseName, null))
+                .Returns(_mockDB.Object);
+
+            //Act
+            var context = new NotificationsContext(_mockSettings.Object); //Create context
+            var userCollection = context.User; //Get collection from context
+
+            //Assert 
+            Assert.NotNull(userCollection);
+        }
+
+        //Tests for user model fields
+        [Fact]
+        public void UserModel_IdTest()
+        {
+            //Arrange
+            var user = new Mock<User>();
+            user.Object.Id = "123123213213";
+
+            //Assert
+            Assert.Equal("123123213213", user.Object.Id);
+        }
```

Comment:
these tests have several problems.
- any new test in a tdd-fail PR is supposed to fail
- mocking the class you're testing
- arrange/act steps misidentified
- but we don't need to unit test POCO classes anyway.